coffee/base.py

```diff
@@ -632,6 +672,32 @@ def itvar(self):
         elif isinstance(self.init, Assign):
             return self.init.children[0]
 
+    def set_itvar(self, new_var):
+        # Change the name of the index init
+        self.init = scopy(self.init)
+        if isinstance(self.init, Decl):
+            self.init.sym.symbol = new_var
+        elif isinstance(self.init, Assign):
+            self.init = Decl("int", Symbol(new_var), self.init.children[1])
+        else:
+            raise NotImplemented("Don't know how to change index name in this init node.")
+        # Change the name of the index in the comparison
```

Comment:
Why are you checking between Symbol and str? str are *not* expected at this point. If there are, that should be fixed and turned into Symbols